python/ray/worker.py

```diff
@@ -198,6 +203,7 @@ class Worker(object):
             that connect has been called already.
         cached_functions_to_run (List): A list of functions to run on all of
             the workers that should be exported as soon as connect is called.
+        reconstruction_lock (Lock): A lock object used to guard object reconstruction.
```

Comment:
In this comment, can you say a little more about what can go wrong without the lock? It's ok to be verbose and to say that when when two threads simultaneously block in ray.get on a worker process, that the calls to self.local_scheduler_client.notify_unblocked will attempt to reacquire the worker's resources, and so it is only safe to call a second self.local_scheduler_client.notify_unblocked after another self.local_scheduler_client.reconstruct_objects has been called.